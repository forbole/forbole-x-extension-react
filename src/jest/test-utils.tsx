import React, { FC, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { ThemeProvider } from '@mui/material';
import lightTheme from '../config/theme/lightTheme';
import '../assets/locales/i18n';

const AllTheProviders: FC = ({ children }) => {
  return <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>;
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
