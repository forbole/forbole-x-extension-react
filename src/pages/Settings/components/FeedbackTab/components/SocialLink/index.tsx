import React from 'react'
import { Link, Box, Typography } from '@mui/material'

type Props = {
  /**
   * The url the link will open a window to.
   */
  URL: string

  /**
   * The svg icon of the link
   */
  Icon: any

  /**
   * The name of the external site the URL redirects to
   */
  siteName: string

  /**
   * The label of the link
   */
  label: string
}

const SocialLink = ({ URL, Icon, label, siteName }: Props) => {
  return (
    <Link
      sx={(_theme) => ({
        color: _theme.palette.text.primary,
        paddingLeft: _theme.spacing(2),
        paddingRight: _theme.spacing(2),
        textDecoration: 'none',
        display: 'flex',
        flex: 1,
        justifyContent: 'space-between',
        alignItems: 'center',
        '&:hover': {
          textDecoration: 'underline',
        },
      })}
      href={URL}
      target="_blank"
      rel="noopener noreferrer"
    >
      <Box
        sx={{
          display: 'flex',
        }}
      >
        <Icon />
        <Typography marginLeft={(_theme) => _theme.spacing(1)} variant="subtitle1">
          {siteName}
        </Typography>
      </Box>

      <Typography>{label}</Typography>
    </Link>
  )
}

export default SocialLink
