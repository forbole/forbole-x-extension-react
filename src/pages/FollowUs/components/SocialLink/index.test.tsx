import React from 'react';
import { render } from 'jest/test-utils';
import ForboleIcon from 'components/svg/ForboleIcon';
import SocialLink from './index';

type Props = React.ComponentProps<typeof SocialLink>;

const testProps: Props = {
  URL: 'https://www.forbole.hk',
  Icon: ForboleIcon,
  siteName: 'Forbole',
  label: 'This is a label',
};

describe('component: SocialLink', () => {
  it('renders', () => {
    const t = render(<SocialLink {...testProps} />);

    expect(t).toMatchSnapshot();
  });

  it('links to URL specified in props', () => {
    const { getByText } = render(<SocialLink {...testProps} />);

    expect(getByText(testProps.label).closest('a')).toHaveAttribute('href', testProps.URL);
  });
});
