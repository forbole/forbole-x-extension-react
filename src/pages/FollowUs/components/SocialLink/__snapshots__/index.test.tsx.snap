// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: SocialLink renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1htsk2q-MuiTypography-root-MuiLink-root"
        href="https://www.forbole.hk"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="MuiBox-root css-k008qs"
        >
          <svg
            fill="#8B8B8B"
            height="24px"
            width="24px"
          >
            icon_forbole.svg
          </svg>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 css-3cy5hm-MuiTypography-root"
          >
            Forbole
          </h6>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body1 css-hi9igu-MuiTypography-root"
        >
          This is a label
        </p>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1htsk2q-MuiTypography-root-MuiLink-root"
      href="https://www.forbole.hk"
      rel="noopener noreferrer"
      target="_blank"
    >
      <div
        class="MuiBox-root css-k008qs"
      >
        <svg
          fill="#8B8B8B"
          height="24px"
          width="24px"
        >
          icon_forbole.svg
        </svg>
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 css-3cy5hm-MuiTypography-root"
        >
          Forbole
        </h6>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body1 css-hi9igu-MuiTypography-root"
      >
        This is a label
      </p>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
